### PR TITLE
fix(TypeScript type definition): Add combineEpics(), provide more accurate type info for others

### LINKED ISSUE #69: [TypeScript] combineEpics is missing in index.d.ts.
I got an TSC error using `redux-observable@0.7.0`

`(6,10): error TS2305: Module '"/path/to/project/node_modules/redux-observable/index"' has no exported member 'combineEpics`

Mostly because it's missing in index.d.ts

### PR DESCRIPTION
**Pull request by jayphelps**
fixes #69

@vire if you get a chance, could you confirm these new typings are good? I don't have any apps that use both TS and redux-observable yet. (though we do have a basic test of the typings included)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,25 +1,17 @@
...
-  // property it is being given the type: `Observable<any>`
-  constructor(actionsSubject: Observable<any>);
+export declare interface Epic {
+  (action$: ActionsObservable, store: MiddlewareAPI<any>): Observable<Action>;
+}
 
-  // ./node_modules/rxjs/Observable.d.ts specifies operators as generics like so
-  // `protected operator: Operator<any, T>;`
-  lift(operator: Operator<any, T>);
+export declare function createEpicMiddleware(...epics: Epic[]): Middleware;
```
Comment (vire) [inline index.d.ts]: I'm a bit unsure about the spread epics here, I thought `createEpicMiddleware` takes just 1 epic (as result of combineEpics) if there are multiple like `createEpicMiddleware(epic1, epic2, epic3)` the epic2 and epic3 are just ignored?

**Comment by vire** [comment]
I've updated my typings definiton from this PR and it works fine.

**Comment by jayphelps** [comment]
@vire Thanks for confirming!

**Comment by jayphelps** [comment]
@vire released as `0.7.1`

File: index.d.ts (hunk 1 above)
Comment (jayphelps) [inline index.d.ts]: Whoops, missed this comment. You're totally right. Going to correct it for a future release.

**Pull request merged**